test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreCommandTests/RestoreCommandTests.cs

```diff
@@ -3190,6 +3190,88 @@ public async Task ExecuteAsync_TransitiveDependenciesFromNonRootLibraries_AreIgn
             }
         }
 
+        [Fact]
+        public async Task ExecuteAsync_WithConditionalProjectAndPackageReferences()
```

Comment:
See naming convention in coding guidelines. Maybe: 
```suggestion
        public async Task ExecuteAsync_WithConditionalProjectAndPackageReferences_WritesLockFileWithCorrectVersions()
```

Reply:
I'll fix that. 
Originally this started as a POC PR with a test that went to NUGet.org. 
I fixed the internals, but I did not address the test name.